docs/tutorial/avax-nft.md

```diff
@@ -0,0 +1,399 @@
+
+## Uploading NFT metadata
+
+Before we can continue, install `nft.storage` and `mime` libraries with `npm install nft.storage mime --save`.
+
+Now we will upload an NFT's metadata -- image, name, and description -- to nft.storage and use the IPFS URI in the `mintTo`. To write a script that can be imported and run on Hardhat Node REPL, create a file called `upload.mjs` inside the `/scripts` directory with the following code (replacing `NFT_STORAGE_KEY` variable with **your own API key**).
+
+```js
+import { NFTStorage, File } from 'nft.storage'
+
+import mime from 'mime'
+
+import fs from 'fs'
+
+import path from 'path'
+
+const NFT_STORAGE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJkaWQ6ZXRocjoweDM5YzIyMUUzOTFiNDMwMzQ4NDc2NzdmMmVGZTc1ODRGNTM2ZjM4OWEiLCJpc3MiOiJuZnQtc3RvcmFnZSIsImlhdCI6MTN0NDU0MjMwNTQ1MSwibmFtZSI6IkF2YWxhbmNoZSJ9.koIFwWwDdhjcBZp2U8OHCiKsfPhXu5aHGXHBQfPXlno'
+
+async function storeNFT(imagePath, name, description) {
+    const image = await fileFromPath(imagePath)
+
+    const nftstorage = new NFTStorage({ token: NFT_STORAGE_KEY })
+
+    return nftstorage.store({
+        image,
+        name,
+        description,
+    })
+}
+
+async function fileFromPath(filePath) {
+    const content = await fs.promises.readFile(filePath)
+    const type = mime.getType(filePath)
+    return new File([content], path.basename(filePath), { type })
+}
+
+async function upload(imagePath, name, description) {
+    const result = await storeNFT(imagePath, name, description)
+    return result
+}
+
+export { upload }
+```
+
+Return to the Node REPL and import the `upload` function from the script. Copy an image file of your choice into the root directory to use as the NFT image.
+
+```js
+>> const { upload } = await import("./scripts/upload.mjs")
+>> const result = await upload("./pickleheart.png", "Pickleheart", "Image of Pickleheart Filet")
+>> result
+> Token {
+>   ipnft: 'bafyreicb3ewk33keh77mwxhmhdafxsjlkflichr2mjnyim6tbq3qjkwcue',
+>   url: 'ipfs://bafyreicb3ewk33keh77mwxhmhdafxsjlkflichr2mjnyim6tbq3qjkwcue/metadata.json'
+> }
+```
+
+Then we can use this new URI in minting:
+
+```js
+>> const tokenId = await filet.callStatic.mintTo(accounts[3], result.url)
+>> tokenId.toNumber()
+> 2
+```
+
+## Retrieving a token's metadata
+
+The final step here is to retrieve the metadata URI for each token from nft.storage so you can display its image, name, and description, which we can do so using [ERC721URIStorage.tokenURI(uint256 tokenId)](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721URIStorage.sol#L20).
+
+```js
+>> let ipfsURI = await filet.tokenURI(tokenId.toNumber())
+>> ipfsURI
+> 'ipfs://bafyreicb3ewk33keh77mwxhmhdafxsjlkflichr2mjnyim6tbq3qjkwcue/metadata.json'
+```
+
+To convert this IPFS URI into an HTTPS version so it's easy to use in HTML or fetch API, you can import a helper function `toGatewayURL` from nft.storage:
```

Comment:
Developers who are not very familiar with IPFS and its browser compatibility may wonder why can't we just use IPFS URI directly. Recommend to add a little explanation.